### PR TITLE
Add autocreate settings

### PR DESCRIPTION
**Pull request by ekohl**
Based on code by @WoLpH in https://github.com/jsocol/django-waffle/pull/45.

### CODE REVIEW

File: waffle/__init__.py [hunk 1]
```diff
@@ -131,12 +136,16 @@ def flag_is_active(request, flag_name):
...
+        if getattr(settings, 'WAFFLE_SWITCH_AUTOCREATE', False):
+            defaults = getattr(settings, 'WAFFLE_SWITCH_DEFAULTS', {})
+            switch, created = Switch.objects.get_or_create(name=switch_name,
+                    defaults=defaults.get(switch_name, {}))
+        else:
+            try:
+                switch = Switch.objects.get(name=switch_name)
+            except Switch.DoesNotExist:
+                switch = DoesNotExist()
+                switch.name = switch_name
```
Comment (ekohl) [inline waffle/__init__.py:148]: This is the odd case which does have negative caching, but I didn't want to modify the existing code.

**Comment by ekohl** [comment]
@jsocol mind having a look?

File: docs/usage.rst [hunk 2]
```diff
@@ -199,6 +199,21 @@ behavior.
     ever upgrade and change the shape of the objects (for example
     upgrading from <0.7.5 to >0.7.5) you'll want to set this to
     something other than ``'waffle:'``.
+``WAFFLE_FLAG_AUTOCREATE``:
+    Whether new Flags will be created automatically when used. Defaults to
+    ``False``.
+``WAFFLE_FLAG_DEFAULTS``:
```
Comment (coagulant) [inline docs/usage.rst:205]: Either WAFFLE_SAMPLE_DEFAULTS is empty dict or it should have defaults under sample name key.

Example:

``` python
WAFFLE_SAMPLE_DEFAULTS = {
    'special_sample': {
        'percent': Decimal(0.15)
    }
}
```

It makes sense, and I guess needs documenting.
Same goes for flags and switches, of course

Comment (ekohl) [inline docs/usage.rst:205]: Made a separate chapter for it now with your example in it.

**Comment by michaelkuty** [comment]
:+1:

**Comment by jberends** [comment]
I am also looking at this PR that implements feats that I need. +1

**Comment by ekohl** [comment]
Closing this because it is indeed stale. Since I never did end up using django-waffle (partly because this PR never got a review) I have no interest in updating it but if others need it then please use it as a basis or inspiration.

**Pull request closed without merging**